rfcs/ils-0104-library-kpis.md

```diff
@@ -0,0 +1,424 @@
+                 - e.g. only count loans that were created manually: filter `loan_creation_method=manually`
+
+     2. Average Waiting Time := waiting time between loan (request) creation and loan start time / number of loans started
+          - query: waiting time between loan (request) creation and loan start time
+            - event: loan starts (use signal `invenio_circulation.signals.loan_state_changed`, indicated by the transition to the `state` `ITEM_ON_LOAN` from any other state)
+            - aggregate:
+              - sum of waiting time
+                - waiting time := loan `start_date` - loan (request) creation date
+                  - loan (request) creation date is based on the creation method := 
+                    - `self_checkout` and `manually`: use the field `_created` on the loan
+                    - `interlibrary_loan`: use either the field
+                      - `_created` on the `invenio_app_ils.borrowing_request` that is linked to the loan
+              - monthly
+              - over field `loan_creation_method__document_availability_during_loan_creation`
+                - see 3.1 for definition of `loan_creation_method__document_availability_during_loan_creation`
+                  - we have to extract the value of this field from the events index of KPI 3.1, as the information is only valid during loan creation 
+                - we also add information about the provider to the event for interlibrary loans, so future aggregations can differentiate the waiting time based on the provider
+            - query: number of loans started
+              - event: loan starts (use signal `invenio_circulation.signals.loan_state_changed`, indicated by the transition to the `state` `ITEM_ON_LOAN` from any other state)
+              - aggregate:
+                - count
+                - monthly
+                - over field `loan_creation_method__document_availability_during_loan_creation`
+                  - see 3.1 for definition of `loan_creation_method__document_availability_during_loan_creation`
+                    - we have to extract the value of this field from the events index of KPI 3.1, as the information is only valid during loan creation 
+                  - we also add information about the provider to the event for interlibrary loans, so future aggregations can differentiate the waiting time based on the provider
+
+
+4. Number of changes to the Library collections:
+    - count number of creations, updates, deletions of all ILS records
+      - The following record types were directly requested by the CERN Library, but we will track all record types:
+        1. documents
+        2. physical items
+        3. e-items
+        4. loans
+      - query: number of creations, updates, deletions of records
+        - event: use existing signals from `invenio_records`
+          - `after_record_insert`
+          - `after_record_update`
+          - `after_record_delete`
+        - aggregate:
```

Comment:
Could you clarify this part - for loans, we already have all the loans indexed with the creation date - why do we need to generate state events? couldn't we just query the loan search and get the answer? is it that because we need it as a number per day?